api/tests/Behat/bootstrap/v2/Analytics/AnalyticsTrait.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\Behat\v2\Analytics;
+
+trait AnalyticsTrait
+{
+    private int $runNumber = 0;
+    private string $currentLinkText = '';
+    private array $expectedMetrics = [];
+    private array $metricXPaths = [
+        'feedBack' => '//span[@aria-labelledby=\'metric-satisfaction-total-label\']',
+        'totalReports' => '//span[@aria-labelledby=\'metric-reportsSubmitted-total-label\']',
+        'totalRegistered' => '//span[@aria-labelledby=\'metric-registeredDeputies-total-label\']',
+        'totalClients' => '//span[@aria-labelledby=\'metric-clients-total-label\']',
+        'laySatisfaction' => '//span[@aria-labelledby=\'metric-satisfaction-deputyType-lay-label\']',
+        'laySubmitted' => '//span[@aria-labelledby=\'metric-reportsSubmitted-deputyType-lay-label\']',
+        'layDeputies' => '//span[@aria-labelledby=\'metric-registeredDeputies-deputyType-lay-label\']',
+        'layClients' => '//span[@aria-labelledby=\'metric-clients-deputyType-lay-label\']',
+        'profSatisfaction' => '//span[@aria-labelledby=\'metric-satisfaction-deputyType-prof-label\']',
+        'profSubmitted' => '//span[@aria-labelledby=\'metric-reportsSubmitted-deputyType-prof-label\']',
+        'profDeputies' => '//span[@aria-labelledby=\'metric-registeredDeputies-deputyType-prof-label\']',
+        'profClients' => '//span[@aria-labelledby=\'metric-clients-deputyType-prof-label\']',
+        'paSatisfaction' => '//span[@aria-labelledby=\'metric-satisfaction-deputyType-pa-label\']',
+        'paSubmitted' => '//span[@aria-labelledby=\'metric-reportsSubmitted-deputyType-pa-label\']',
+        'paDeputies' => '//span[@aria-labelledby=\'metric-registeredDeputies-deputyType-pa-label\']',
+        'paClients' => '//span[@aria-labelledby=\'metric-clients-deputyType-pa-label\']',
+    ];
```

Comment:
Nice - I've slowly grown to ~love~ like xpath through this whole behat shizzle

Reply:
Haha love is a strong word but it's actually quite good. Was experimenting using ancestor before which might have some use case... just turns out i didn't need it :)